lib/board.rb

```diff
@@ -0,0 +1,81 @@
+require 'pry'
+
+
+class Board
+
+  attr_accessor :cells, :token, :move, :player
+
+ 
+
+    def initialize
+      @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    end
+
+
+
+    def reset!
+      @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    end
+
+    def display
+      puts " ----------- "
+      puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+      puts " ----------- "
+      puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+      puts " -----------"
+      puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    end   
+
+    def position(input)
+      move = input.to_i
+      move -= 1
+      @cells[move]
+    end
+
+    def full?
+      true if !@cells.detect {|x| x == " "}
+    end
+
+
+
+    def turn_count
+      counter = 0
+      @cells.each do |x|
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count { |cell| cell == "X" || cell == "O" }
end
```